app/controllers/sessions.rb

```diff
@@ -0,0 +1,21 @@
+get '/sessions/new' do
+  erb :"sessions/new"
+end
+
+post '/sessions' do
+  username, password = params[:username], params[:password]
```

Comment:
you are only using username and password here once, you could put the params directly into authenticate to DRY this out